code/modules/mob/living/silicon/ai/ai_programs.dm

```diff
@@ -0,0 +1,998 @@
+	user.playsound_local(user, 'sound/effects/spray.ogg', 50, FALSE, use_reverb = FALSE)
+	playsound(target, 'sound/effects/spray.ogg', 50, FALSE, use_reverb = FALSE)
+	var/obj/machinery/camera/C = find_nearest_camera(target)
+	if(!istype(C))
+		return
+	C.Beam(target, icon_state = "rped_upgrade", icon = 'icons/effects/effects.dmi', time = 5)
+
+/datum/spell/ai_spell/ranged/rgb_lighting/on_purchase_upgrade()
+	cooldown_handler.recharge_duration = base_cooldown - (spell_level * 5)
+
+// Power Shunt - Recharges things from your SMES
+/datum/ai_program/power_shunt
+	program_name = "Power Shunt"
+	program_id = "power_shunt"
+	description = "Recharge an APC, Borg, or Mech with power directly from your SMES!"
+	nanite_cost = 20
+	power_type = /datum/spell/ai_spell/ranged/power_shunt
+	unlock_text = "Power Shunt installation complete!"
+
+/datum/spell/ai_spell/ranged/power_shunt
+	name = "Power Shunt"
+	desc = "Recharge an APC, Borg, or Mech with power directly from your SMES"
+	action_icon = 'icons/obj/power.dmi'
+	action_icon_state = "smes-o"
+	ranged_mousepointer = 'icons/mecha/mecha_mouse.dmi'
+	auto_use_uses = FALSE
+	base_cooldown = 300 SECONDS
+	cooldown_min = 30 SECONDS
+	level_max = 7
+	selection_activated_message = "<span class='notice'>You tap into the station's powernet...</span>"
+	selection_deactivated_message = "<span class='notice'>You release your hold on the powernet.</span>"
+	var/power_sent = 2500
+
+/datum/spell/ai_spell/ranged/power_shunt/cast(list/targets, mob/user)
+	var/target = targets[1]
+	if(!istype(target, /mob/living/silicon/robot) && !istype(target, /obj/machinery/power/apc) && !istype(target, /obj/mecha) && !istype(target, /mob/living/carbon/human/machine))
+		to_chat(user, "<span class='warning'>You can only recharge borgs, mechs, and APCs!</span>")
+		return
+	var/mob/living/silicon/ai/AI = user
+	if(istype(target, /mob/living/carbon/human/machine)  && !AI.universal_adapter)
+		to_chat(user, "<span class='warning'>This software lacks the required upgrade to recharge IPCs!</span>")
+		return
+	var/area/A = get_area(user)
+	if(A == null)
+		to_chat(user, "<span class='warning'>No SMES detected to power from!</span>")
+		return
+	if(istype(target, /obj/mecha))
+		var/obj/mecha/T = target
+		T.cell.give(power_sent)
+	if(istype(target, /mob/living/silicon/robot))
+		var/mob/living/silicon/robot/T = target
+		T.cell.give(power_sent)
+	if(istype(target, /obj/machinery/power/apc))
+		var/obj/machinery/power/apc/T = target
+		T.cell.give(power_sent)
+	if(istype(target, /mob/living/carbon/human/machine))
+		var/mob/living/carbon/human/machine/T = target
+		T.adjust_nutrition(AI.adapter_efficiency * (power_sent / 10))
+
+	// Handles draining the SMES
+	for(var/obj/machinery/power/smes/power_source in A)
+		if(power_source.charge < power_sent)
+			continue
+		power_source.charge -= power_sent
+		break
+	AI.program_picker.nanites -= 20
+	user.playsound_local(user, 'sound/goonstation/misc/fuse.ogg', 50, FALSE, use_reverb = FALSE)
```

Comment:
Might not be a bad idea to add a helper for AIs that combines the two sound playing calls